### PR TITLE
Feature is lb_listeners/lb_listener, lb_listener_policies/lb_listener_policy, lb_listener_policy_rule/lb_listener_policy_rules

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
Screenshot 2021-11-29 at 3 53 07 PM
Screenshot 2021-11-29 at 3 53 18 PM
Screenshot 2021-11-29 at 3 53 28 PM
Screenshot 2021-11-29 at 3 55 13 PM
Screenshot 2021-11-29 at 4 11 42 PM

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by SunithaGudisagarIBM** [comment]
Screenshot 2021-11-30 at 3 37 02 PM
Screenshot 2021-11-30 at 12 38 35 PM
Screenshot 2021-11-30 at 12 51 04 PM
Screenshot 2021-12-01 at 5 22 10 PM

File: ibm/data_source_ibm_is_lb_listeners_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsLbListenersDataSourceConfigBasic(vpcname, subnetname, ISZoneName, ISCIDR, lbname, port1, protocol1),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_is_lb_listeners.is_lb_listeners", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_lb_listeners.is_lb_listeners", "lb"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_lb_listeners.is_lb_listeners", "listeners.#"),
```
Comment (BShrivastav) [inline ibm/data_source_ibm_is_lb_listeners_test.go:29]: Could we please add the test support for the child attributes which are mandatory.

File: ibm/data_source_ibm_is_lb_listener_policies.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+	deletedMap = map[string]interface{}{}
+
+	if deletedItem.MoreInfo != nil {
+		deletedMap["more_info"] = deletedItem.MoreInfo
+	}
+
+	return deletedMap
+}
+
+func dataSourceLoadBalancerListenerPolicyCollectionPoliciesTargetToMap(targetItem vpcv1.LoadBalancerListenerPolicyTarget) (targetMap map[string]interface{}) {
```
Comment (BShrivastav) [inline ibm/data_source_ibm_is_lb_listener_policies.go:342]: Could we consider the oneof check!
https://pages.github.ibm.com/riaas/api-spec/spec_genesis_2021-12-07/#:~:text=LoadBalancerListenerPolicyTarget,LoadBalancerListenerPolicyTarget

File: website/docs/d/is_lb_listener.html.markdown [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	listener_id = ibm_is_lb_listener.example.listener_id
+	lb = ibm_is_lb.example.id
+}
+```
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+- `listener_id` - (Required, String) The listener identifier.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener.html.markdown:25]: Can we remove the Required/Optional.

File: website/docs/d/is_lb_listener_policies.html.markdown [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+- `listener_id` - (Required, String) The listener identifier.
+- `load_balancer_id` - (Required, String) The load balancer identifier.
+- `name` - (Optional, String) The user-defined name for this policy.
+
+## Attribute Reference
+
+In addition to all argument references listed, you can access the following attribute references after your data source is created.
+
+- `id` - The unique identifier of the LoadBalancerListenerPolicyCollection.
+- `policies` - (Required, List) Collection of policies.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener_policies.html.markdown:35]: Could we remove the Required/Optional.

File: website/docs/d/is_lb_listener_policies.html.markdown [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	listener = ibm_is_lb_listener.example.listener_id
+	name = ibm_is_lb_listener_policy.example.name
+}
+```
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+- `listener_id` - (Required, String) The listener identifier.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener_policies.html.markdown:26]: This needs to be aligned to the right name, as per the above example.

File: website/docs/d/is_lb_listener_policy_rules.html.markdown [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	lb = ibm_is_lb.example.id
+	policy = ibm_is_lb_listener_policy.example.policy_id
+}
+```
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+- `listener_id` - (Required, Forces new resource, String) The listener identifier.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener_policy_rules.html.markdown:25]: Correction in the name as per the actual parameters.

File: website/docs/d/is_lb_listener_policy_rules.html.markdown [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+## Attribute Reference
+
+In addition to all argument references listed, you can access the following attribute references after your data source is created.
+
+- `id` - The unique identifier of the LoadBalancerListenerPolicyRuleCollection.
+- `rules` - (Required, List) Collection of rules.
+Nested scheme for `rules`:
+	- `condition` - (Required, String) The condition of the rule.
+	- `created_at` - (Required, String) The date and time that this rule was created.
+	- `field` - (Optional, String) The field. This is applicable to `header`, `query`, and `body` rule types.If the rule type is `header`, this property is required.If the rule type is `query`, this is optional. If specified and the rule condition is not`matches_regex`, the value must be percent-encoded.If the rule type is `body`, this is optional.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener_policy_rules.html.markdown:38]: Required/Optional, please remove if not needed.

File: website/docs/d/is_lb_listener_policy_rule.html.markdown [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+- `listener` - (Required, String) The listener identifier.
+- `lb` - (Required, String) The load balancer identifier.
+- `policy` - (Required, String) The policy identifier.
+
+## Attribute Reference
+
+In addition to all argument references listed, you can access the following attribute references after your data source is created.
+
+- `id` - The unique identifier of the LoadBalancerListenerPolicyRule.
+- `condition` - (Required, String) The condition of the rule.
```
Comment (BShrivastav) [inline website/docs/d/is_lb_listener_policy_rule.html.markdown:37]: Required/Optional please remove as required.

File: ibm/data_source_ibm_is_lb_listener.go [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+		return diag.FromErr(fmt.Errorf("Error setting accept_proxy_protocol: %s", err))
+	}
+
+	if loadBalancerListener.CertificateInstance != nil {
+		err = d.Set("certificate_instance", dataSourceLoadBalancerListenerFlattenCertificateInstance(*loadBalancerListener.CertificateInstance))
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting certificate_instance %s", err))
+		}
+	}
+	if err = d.Set("connection_limit", intValue(loadBalancerListener.ConnectionLimit)); err != nil {
```
Comment (BShrivastav) [inline ibm/data_source_ibm_is_lb_listener.go:245]: Optional field:

```
if loadBalancerListener.ConnectionLimit != nil {
		if err = d.Set("connection_limit", intValue(loadBalancerListener.ConnectionLimit)); err != nil {
			return diag.FromErr(fmt.Errorf("Error setting connection_limit: %s", err))
		}
	}
```

File: website/docs/d/is_lb_listener_policy_rules.html.markdown (hunk 6 above)
Comment (SunithaGudisagarIBM) [reply website/docs/d/is_lb_listener_policy_rules.html.markdown:25]: Can you check again now please

File: ibm/data_source_ibm_is_lb_listener_policies.go (hunk 2 above)
Comment (SunithaGudisagarIBM) [reply ibm/data_source_ibm_is_lb_listener_policies.go:342]: hi.. i didnt get you here

**Pull request closed without merging**